File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/G. Factorial/main.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/G
#include<iostream>
#include<cstdio>

using namespace std;
 
int main(){
    unsigned short t, n;
    cin >> t;
    while(t--){
        cin >> n;
        unsigned long long fact = n !=0 ? n : 1;
        for(int i=1; i<n; i++){
            fact *= i;
        }
        cout << fact << endl;
    }
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/G. Factorial/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/G

using System;

namespace Codeforces.Assuit.Contest1
{
    class ProblemG
    {
        static void Main(string[] args)
        {
            ushort t, n;
            t = ushort.Parse(Console.ReadLine());
            while(t-- > 0){
                n = ushort.Parse(Console.ReadLine());
                ulong fact = (ulong)(n != 0 ? n : 1);
                for(int i=1; i<n; i++){
                    fact *= (ulong)i;
                }
                Console.WriteLine(fact);
            }
        }
    }
}
